src/backend/dx12/src/device.rs

```diff
@@ -1496,11 +1496,104 @@ impl d::Device<B> for Device {
 
         // Create PSO
         let mut pipeline = ptr::null_mut();
-        let hr = unsafe {
-            self.raw.clone().CreateGraphicsPipelineState(
-                &pso_desc,
-                &d3d12::IID_ID3D12PipelineState,
-                &mut pipeline as *mut *mut _ as *mut *mut _)
+        let hr = if desc.depth_stencil.unwrap_or(unsafe { mem::zeroed() }).depth_bounds {
```

Comment:
nit: maps/unwrap_or is most useful when you have linear code. Once you need to process both branches, a simple `match` looks better (to me):
```rust
let hr = match desc.depth_stencil {
   Some(ds) if ds.depth_bounds => {...}
   _ => {...}
};
```

Reply:
I thought about it but the `None` case is the same as the `false` case and I wasn't sure howto combine the two.